gigascience/gigadb-website
# Admin forms layout fills screen width

# Pull request for issue: #2100

IMPORTANT: #2168 extends this PR by updating all the forms to the well layout type. We might merge this PR if we prefer the unwrapped layout or merge #2168 if we prefer the wrapped layout

if preferred layout = unwrapped:
- merge this PR
- close PR #2168

if preferred layout = wrapped
- merge PR #2168
- close this PR

Unwrapped layout (this PR)
![image](https://github.com/user-attachments/assets/703808cb-385b-4892-be0a-24fa84882e9c)

Wrapped layout (#2168)
![image](https://github.com/user-attachments/assets/8aeded42-3de0-4308-a7e1-9cf9ed6bcfcb)

As per the comments in #2100 the wrapped layout is preferred

Updated all admin forms to make use of a wider region of the page

## How to test?

- there are no changes in the html and functionality so all existing tests should pass
- navigate to each admin form (except dataset) and verify layout now fills the horizontal space available

## How have functionalities been implemented?

- used different bootstrap classes to use wider responsive layouts
- on mobile, layout becomes a column
- field widths were weighted taking into account visual balance and field type, varying between one, two and three columns, depending on the case
- I opted to add an additional button style to keep button widths more consistant

## Any issues with implementation?

there is no way to apply these layout changes consistently while keeping all forms looking great, there are forms with just 2 or 3 fields which leave blank space in the page no matter what layout is adopted

I still think a wide layout looks better generally speaking

## Any changes to automated tests?

All existing tests pass
